src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -197,4 +205,19 @@ export class ClusterClient {
       throw new Error('Cluster client cannot be used after it has been closed.');
     }
   }
+
+  // supports both KibanaRequest & Legacy.Request while asScoped is used in LegacyPlatform
+  private getHeaders(
+    request?: KibanaRequest | Request
+  ): Record<string, string | string[] | undefined> {
+    if (!request) return {};
```

Comment:
BWC with the current code. IMO it's weird that `cluster.asScoped(undefined)` is a valid use case.